javalib/src/main/scala/java/io/File.scala

```diff
@@ -415,7 +415,7 @@ class File(_path: String) extends Serializable with Comparable[File] {
       } else {
         val buf = alloc[stat.stat]()
         if (stat.stat(toCString(path), buf) == 0) {
-          buf._8 * 1000L
+          buf._8._1 * 1000L
```

Comment:
When we get the `statOps` introduced below to smartly
handle 32/64 bit fields(st_atime /st_atim, this is and its relatives are  
textbook candidates, for all the reasons you created that
class.

`st_atime` is a drop in replacement, for now.  I think you 
hinted at extended at another PR which will extend javalibe to handle sub-second where the Java 8 spec allows implementation defined. (I am looking forward to both this and that PR).